pandas/tseries/offsets.py

```diff
@@ -97,6 +97,8 @@ def wrapper(self, other):
             if tz is not None and result.tzinfo is None:
                 result = conversion.localize_pydatetime(result, tz)
 
+            result = Timestamp(result)
```

Comment:
why does this not just raise again?

Reply:
Without this we incorrectly return a `datetime` object instead of a `Timestamp` in `generate_range`, and `test_bday_overflow_error` when it tries to do `[x.value for x in thing_we_got_from_generate_range]`